Task/WG-260 add map control bar

nathanfranklin opened the pull request:
## Overview: ##

This PR introduces a control bar at the top of the map to align with the behavior of the Angular Hazmapper app:
- Show map name
- Displays the associated DesignSafe project name.
- Shows a spinner while features are being loaded.

It also handles different errors that could occur.

Additional Hooks:

- useMapMousePosition: Tracks the mouse position on the map.
- useCurrentFeatures: Updated to indicate when new features are loading.
- useProjectUsers: Retrieves map project users.

## PR Status: ##

* [X] Ready.

## Related Jira tickets: ##

* [WG-260](https://tacc-main.atlassian.net/browse/WG-260)

## Summary of Changes: ##

## Testing Steps: ##

Example routes are provided that correspond to maps on staging (i.e. `GeoapiBackendEnvironment.Staging` in `src/secret_local.ts` )

-  http://localhost:4200/project/cd010f4d-3975-4fde-8bbd-b81ffb87273f
   - confirm shows map name and project name
   - show loading when features are requested (occurs very quickly on this small map).
- http://localhost:4200/project-public/cd010f4d-3975-4fde-8bbd-b81ffb87273f
   -  see public view of map
   -  could go into incognito to see this view as well.
- http://localhost:4200/project/b148ab1c-3195-49d9-81cc-da845bae59d4.  (made up incorrect map) 
   -  see `This map project does not exist`;   right now real uuids have to be used but fixing in https://github.com/TACC-Cloud/geoapi/pull/236 
- http://localhost:4200/project/47d38caf-c800-468a-91db-f905cc661ccb (map only for user `nathan_f`
   -  see 'You don't have permission to access this map project'

## UI Photos:

-----

Error getting map:
![Screenshot 2025-01-21 at 11 22 55 AM](https://github.com/user-attachments/assets/c7328769-6682-47cc-a8d3-28d0c18bada5)

------------------------------

Map doesn't exist 

![Screenshot 2025-01-21 at 4 21 35 PM](https://github.com/user-attachments/assets/bcb5f45e-017f-46a4-8dd9-48bd3a1fb3b6)

---------------

Map is private and user isn't a member of the project:

![Screenshot 2025-01-21 at 4 26 18 PM](https://github.com/user-attachments/assets/e30f9bac-ef17-4dd8-9b9c-f736316caf51)


-----------------------------
Public map (not logged in)
![Screenshot 2025-01-21 at 11 22 23 AM](https://github.com/user-attachments/assets/8fbf13ed-73af-45fe-a657-d225cf0d03b1)
-----------------------------

Public map (logged in)

![Screenshot 2025-01-21 at 11 21 44 AM](https://github.com/user-attachments/assets/56349368-3d02-4f34-adc3-8bbf09e5bc3f)
-----------------------------

Private map view
![Screenshot 2025-01-21 at 11 21 35 AM](https://github.com/user-attachments/assets/e089f820-122d-4851-891d-c8ddf02f8ffe)


## Notes: ##

sophia-massie reviewed (approved):
Tested maps and LGTM. 
Not critical but should the MapControlBar have a larger font like prod?

Follow on:
I'm seeing an issue for a public map where the Add Asset from DesignSafe button is appearing when it should not. [See here](http://localhost:4200/project-public/cd010f4d-3975-4fde-8bbd-b81ffb87273f?panel=Assets&selectedFeature=2471910) Not a bug from your code, but the handling of if the AssetButton should show or not is faulty.

sophia-massie reviewed (approved):
LGTM!

nathanfranklin commented:
> Tested maps and LGTM. Not critical but should the MapControlBar have a larger font like prod?
Added note for this to look at in Feb

> Follow on: I'm seeing an issue for a public map where the Add Asset from DesignSafe button is appearing when it should not. [See here](http://localhost:4200/project-public/cd010f4d-3975-4fde-8bbd-b81ffb87273f?panel=Assets&selectedFeature=2471910) a bug from your code, but the handling of if the AssetButton should show or not is faulty.

fixed in separate PR.
